### ISSUE TITLE
Implementation questions

### ISSUE THREAD

**Issue by checkpoint214159** [bair-climate-initiative/scale-mae#19]
Hi, thanks for the work you guys have put into this research. However, I do have a few pressing questions I cannot find any solutions to, and was hoping somebodoy could give some insight.

1. What was the positional encoding added to the encoder outputs? My impression is that it was a GSD positional encoding (GSD PE) using the target resolution, not the input resolution. (e.g from Fig 2 in the paper, input_resolution is 0.7m, target_resolution is 0.3m, correct me if I am mistaken?) However, looking at the code, it seems to be calculated using the input resolution of 0.7m. I believed that, for the decoder to produce an accurate reconstruction of the target, the GSD PE should be calculated from **_the target resolution_**, not from **_the input resolution_** as was done in the code. Is this the correct implementation?

2. Furthermore, looking into the forward method of the code, specifically at line 362 of scale-mae/mae/models_mae.py of the main branch, I see
`pos_embed = get_2d_sincos_pos_embed_with_resolution(
            x.shape[-1], target_dim, target_res, cls_token=True, device=x.device
        )`

Which is then projected under some embedding projection module, but never used for anything else. Is this intentional, or am I just unable to find where it is used? This is one of the reasons why I asked question 1.; it seems like the intention was to get GSD PE with target resolution encoding, but the implementation never showed as such.

As a whole, I would also like to comment that better documentation in place for the backend code, would make it less confusing to traverse, as even having been through the code from the SatMAE paper which this was built upon, I felt loss quite a few times. Regardless, thank you for your hard work and dedication in open-sourcing and publishing your research :)

**Comment by HappyGoLucky6666** [comment]
Hi, can you please answer some of my questions about this program? I have recently encountered difficult difficulties and unintelligible doubts while reproducing the project, and I can sense from your comments that you seem to have understood the project. First of all, the first point is about the pre-training configuration file of this project, is the config.yaml parameter specified in the pre-training main_pretrain file set according to the actual demand? How to set your own customized file?@checkpoint214159

**Comment by checkpoint214159** [comment]
Hi, sure, as context I was surveying the field of pre-training for satellite imagery for an internship, came across this paper and github, and felt curious enough to try it out. We were developing a pipeline for pre-training / finetuning foundational models

**Comment by checkpoint214159** [comment]
Hi, sure, as context I was surveying the field of pre-training for satellite imagery for an internship that recently ended, came across this paper and github, and felt curious enough to try it out. We were developing a pipeline for pre-training / finetuning foundational models in MMEngine, and I focused mainly on how to adapt the model class to our own. Thus, to answer your question, I did not use their configuration system and instead used our own (built off MMConfig). 

Additionally, as I did feel rather confused by the model implementation and had some trouble porting it to our pipeline, after spending considerable time we ended up leaving behind ScaleMAE and did not continue experimenting with it. This is not to say anything about this project; all it is, is that we decided on other priorities that did not include continuing to adapt this to our framework.

As an alternative, I would direct you to SatMAE or Swin-MAE for other masked image reconstruction pre-training architectures that we did implement and try out. These were easier to work with and produce reasonable outcomes / converge on our own datasets. Notably, we found for our case, Swin-MAE's architecture works considerably better for dense prediction tasks, especially for the domain of satellite imagery, so you might be interested to look into that.

**Comment by HappyGoLucky6666** [comment]
Thank you for your reply, and I will listen carefully to your suggestions. I was looking forward to this project, but I can't reproduce it, the configuration of this project is not yet understood, it's complicated, maybe I'll give up strategically and work on other models after some time.

**Comment by sauryeo** [comment]
Hello, sorry to bother you. I saw your comment on this piece of code and would like to ask you a few questions.

I'm currently trying to fine-tune the model on a downstream task using the pretrained weights provided by the authors. Do you think the code issue you mentioned would affect the fine-tuning results of downstream tasks?

Additionally, I'm interested in studying the SatMAE model. I noticed that the pretraining dataset used in the paper is temporal. In your reproduction, did you use the official dataset or a custom one? I would like to replace it with my own dataset (which is not temporal). Could you advise me on how to handle that?

Looking forward to your response.
